typespec-extension/src/code-model-builder.ts

```diff
@@ -736,73 +715,66 @@ export class CodeModelBuilder {
           ).toString();
         }
 
-        operationExamples[operationExample.title ?? operationExample.operationId ?? operation.operation.name] =
-          operationExample;
+        operationExamples[operationExample.title ?? operationExample.operationId ?? sdkMethod.name] = operationExample;
       }
       return operationExamples;
     } else {
       return undefined;
     }
   }
 
-  private processOperation(groupName: string, operation: Operation, clientContext: ClientContext): CodeModelOperation {
-    const op = getHttpOperationWithCache(this.sdkContext, operation);
-
+  private processOperation(
+    sdkMethod: SdkServiceMethod<SdkHttpOperation>,
+    clientContext: ClientContext,
+    groupName: string,
+  ): CodeModelOperation {
+    const operationName = sdkMethod.name;
+    const httpOperation = sdkMethod.operation;
+    const operationId = groupName ? `${groupName}_${operationName}` : `${operationName}`;
     const operationGroup = this.codeModel.getOperationGroup(groupName);
-    const operationName = this.getName(operation);
-    const opId = groupName ? `${groupName}_${operationName}` : `${operationName}`;
 
-    const operationExamples = this.getOperationExample(op);
+    let operationExamples = undefined;
+    if (sdkMethod.operation && sdkMethod.operation.__raw) {
+      operationExamples = this.getOperationExample(sdkMethod);
+    }
 
-    const codeModelOperation = new CodeModelOperation(operationName, this.getDoc(operation), {
-      operationId: opId,
-      summary: this.getSummary(operation),
+    const codeModelOperation = new CodeModelOperation(operationName, sdkMethod.details ?? "", {
+      operationId: operationId,
+      summary: sdkMethod.description,
       extensions: {
         "x-ms-examples": operationExamples,
       },
     });
 
-    (codeModelOperation as CrossLanguageDefinition).crossLanguageDefinitionId = getCrossLanguageDefinitionId(
-      this.sdkContext,
-      operation,
-    );
-    codeModelOperation.internalApi = this.isInternal(operation);
+    (codeModelOperation as CrossLanguageDefinition).crossLanguageDefinitionId = sdkMethod.crossLanguageDefintionId;
+    codeModelOperation.internalApi = sdkMethod.access === "internal";
 
-    const convenienceApiName = this.getConvenienceApiName(operation);
+    const convenienceApiName = this.getConvenienceApiName(sdkMethod);
     let generateConvenienceApi: boolean = Boolean(convenienceApiName);
-    let generateProtocolApi: boolean = shouldGenerateProtocol(this.sdkContext, operation);
+    let generateProtocolApi: boolean = sdkMethod.__raw
+      ? shouldGenerateProtocol(this.sdkContext, sdkMethod.__raw)
+      : true;
```

Comment:
net, it should be more natural to put the `sdkMethod.__raw` check into `shouldGenerateProtocol`.